scripts/globals/spells/sleepga.lua

```diff
@@ -17,8 +17,20 @@ function onSpellCast(caster,target,spell)
     local mINT = target:getStat(MOD_INT);
     local dINT = (pINT - mINT);
     local resm = applyResistanceEffect(caster,spell,target,dINT,ENFEEBLING_MAGIC_SKILL,0,EFFECT_SLEEP_I);
+
+    if (caster:isMob()) then
+        if (caster:getPool() == 5310) then -- Amnaf (Flayer)
+            caster:resetEnmity(target);
+        end
+        -- Todo: get rid of this whole block by handling it in the mob script
+        -- this requires a multi target enmity without specifying a target (have to get hate list from mob)
+        -- OR by altering onSpellPrecast to have a target param..
+        -- onMonsterMagicPrepare is not a realistic option.
+        -- You'd have to script the use of every individual spell in Amnaf's list..
+    end
+
```

Comment:
i had a mob:clearAllEnmity() in this branch, but it was reverted a few commits back with the disengage changes